encoder-brave/src/main/java/zipkin2/reporter/otel/brave/SpanTranslator.java

```diff
@@ -0,0 +1,212 @@
+  static final SpanKind DEFAULT_KIND = SpanKind.SPAN_KIND_INTERNAL;
+
+  /**
+   * Tag to Attribute mappings which map brave data policy to otel semantics.
+   *
+   * @see <a href="https://opentelemetry.io/docs/specs/semconv/http/http-spans/">https://opentelemetry.io/docs/specs/semconv/http/http-spans/</a>
+   * @see brave.http.HttpTags
+   */
+  // TODO: brave also defines rpc and messaging data policy
+  static final Map<String, String> RENAMED_ATTRIBUTES;
+
+  static {
+    RENAMED_ATTRIBUTES = new LinkedHashMap<>();
+    // "http.host" is not defined in HttpTags, but is a well-known tag.
+    RENAMED_ATTRIBUTES.put("http.host", ServerAttributes.SERVER_ADDRESS.getKey());
+    RENAMED_ATTRIBUTES.put(HttpTags.METHOD.key(), HttpAttributes.HTTP_REQUEST_METHOD.getKey());
+    RENAMED_ATTRIBUTES.put(HttpTags.PATH.key(), UrlAttributes.URL_PATH.getKey());
+    RENAMED_ATTRIBUTES.put(HttpTags.ROUTE.key(), HttpAttributes.HTTP_ROUTE.getKey());
+    RENAMED_ATTRIBUTES.put(HttpTags.URL.key(), UrlAttributes.URL_FULL.getKey());
+    RENAMED_ATTRIBUTES.put(HttpTags.STATUS_CODE.key(), HttpAttributes.HTTP_RESPONSE_STATUS_CODE.getKey());
+  }
+
+  private final Consumer consumer;
+
+  SpanTranslator(Tag<Throwable> errorTag) {
+    this.consumer = new Consumer(errorTag, RENAMED_ATTRIBUTES);
+  }
+
+  TracesData translate(MutableSpan braveSpan) {
+    TracesData.Builder tracesDataBuilder = TracesData.newBuilder();
+    Builder resourceSpansBuilder = ResourceSpans.newBuilder();
+    ScopeSpans.Builder scopeSpanBuilder = ScopeSpans.newBuilder();
+    Span.Builder spanBuilder = builderForSingleSpan(braveSpan, resourceSpansBuilder);
+    scopeSpanBuilder.addSpans(spanBuilder
+        .build());
+    InstrumentationScope.Builder scopeBuilder = InstrumentationScope.newBuilder();
+    scopeBuilder.setName(BraveScope.getName());
+    scopeBuilder.setVersion(BraveScope.getVersion());
+    scopeSpanBuilder.setScope(scopeBuilder.build());
+    resourceSpansBuilder.addScopeSpans(scopeSpanBuilder
+        .build());
+    tracesDataBuilder.addResourceSpans(resourceSpansBuilder.build());
+    return tracesDataBuilder.build();
+  }
+
+  static KeyValue stringAttribute(String key, String value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setStringValue(value))
+        .build();
+  }
+
+  static KeyValue intAttribute(String key, int value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setIntValue(value))
+        .build();
+  }
+
+  private Span.Builder builderForSingleSpan(MutableSpan span, Builder resourceSpansBuilder) {
+    Span.Builder spanBuilder = Span.newBuilder()
+        .setTraceId(span.traceId() != null ? ByteString.fromHex(span.traceId()) : INVALID_TRACE_ID)
+        .setSpanId(span.id() != null ? ByteString.fromHex(span.id()) : INVALID_SPAN_ID);
+    if (span.name() != null) {
```

Comment:
If we can remove this check that would be good. If we need to do it defensively, we should set to a default like `unknown`. The field is required 

https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry%2Fproto%2Ftrace%2Fv1%2Ftrace.proto#L145

Reply:
If this `if` statement is remove, the name will be missing if `span.name()` is `null`. On the other hand, if this `if` statement exists, the span name will be set to an empty string if `span.name()` is `null` (the default value of the SDK?).
Do you intend this behavior?

Reply:
With protos, when a field has a comment that it's required it means it needs to be a non-empty string since it doesn't have a concept of null. So we should ensure the field is set to something.

I don't know if there's any realistic scenario where name could be none in brave so in that case it seems fine to just remove the if and rely on the assumption. If there is a scenario it happens though then I think we should set to a non-empty string in an `else` case.

Reply:
Thanks!　Change the default value of span to `unknown` in https://github.com/openzipkin-contrib/zipkin-otel/pull/9/commits/016fac06061558c73da09ab1ab56d3ba9af69515